MediaBrowser.Model/FileOrganization/EpisodeFileOrganizationRequest.cs

```diff
@@ -13,5 +15,38 @@ public class EpisodeFileOrganizationRequest
         public int? EndingEpisodeNumber { get; set; }
 
         public bool RememberCorrection { get; set; }
+        public string NewSeriesName { get; set; }
+
+        public string NewSeriesYear { get; set; }
+
+        public string NewSeriesProviderIds { get; set; }
+
+        public string TargetFolder { get; set; }
+
+        public Dictionary<string, string> NewSeriesProviderIdsDictionary
```

Comment:
I kept the NewSeriesName and NewSeriesYear separate in the API for the following reasons:
- A future implementation (or different client using the API) might want to allow explicit specification of the series folder name. Use cases may be trivial like to have "The Good Wife" instead of "Good Wife" but also language related. A series might have been identified as "Crime Scene Cleaner", but the user wants to use the original name "Der Tatortreiniger"
- IMO an API should provide a reasonable amount of flexibility and not be arbitrarily limited exactly to the use case proposed at the time of creation

If my arguments still didn't convince, I'd like to remark that reducing the signature to nothing but the provider ids would require an additional metadata query at the server, because the current process is like this:
- series = new Series();
- series.Name = request.NewSeriesName;
- series.Path = Path.Combine(request.TargetFolder, seriesFolderName);
- series.ProviderIds = request.NewSeriesProviderIdsDictionary;
- series.RefreshMetadata(refreshOptions, cancellationToken);

So, the name is required before metatada is refreshed.

On the contrary, if we would really pass the provider ids only, there is no 100% guarantee that the query by provider ids will result the exact same item that the user had searched by name in the UI!